// ol-react/lib/map.js
"use strict";

var _interopRequire = function (obj) { return obj && obj.__esModule ? obj["default"] : obj; };

var _createClass = (function () { function defineProperties(target, props) { for (var key in props) { var prop = props[key]; prop.configurable = true; if (prop.value) prop.writable = true; } Object.defineProperties(target, props); } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; })();

var _get = function get(object, property, receiver) { var desc = Object.getOwnPropertyDescriptor(object, property); if (desc === undefined) { var parent = Object.getPrototypeOf(object); if (parent === null) { return undefined; } else { return get(parent, property, receiver); } } else if ("value" in desc && desc.writable) { return desc.value; } else { var getter = desc.get; if (getter === undefined) { return undefined; } return getter.call(receiver); } };

var _inherits = function (subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) subClass.__proto__ = superClass; };

var _classCallCheck = function (instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } };

var React = _interopRequire(require("react"));

var ol = _interopRequire(require("openlayers"));

var OLComponent = _interopRequire(require("./ol-component"));

var Map = (function (_React$Component) {
  function Map(props) {
    _classCallCheck(this, Map);

    _get(Object.getPrototypeOf(Map.prototype), "constructor", this).call(this, props);
    this.map = new ol.Map({
      loadTilesWhileAnimating: props.loadTilesWhileAnimating,
      loadTilesWhileInteracting: props.loadTilesWhileInteracting,
      interactions: props.useDefaultInteractions ? ol.interaction.defaults() : [],
      controls: props.useDefaultControls ? ol.control.defaults() : []
    });

    if (props.onChangeSize) {
      this.map.on("change:size", props.onChangeSize);
    }
    if (this.props.onSingleClick) {
      this.map.on("singleclick", this.props.onSingleClick);
    }
  }

  _inherits(Map, _React$Component);

  _createClass(Map, {
    componentDidMount: {
      value: function componentDidMount() {
        this.map.setTarget(this.refs.target);

        if (this.props.focusOnMount) {
          this.focus();
        }
      }
    },
    componentWillUnmount: {
      value: function componentWillUnmount() {
        this.map.setTarget(undefined);
      }
    },
    getChildContext: {
      value: function getChildContext() {
        return {
          map: this.map
        };
      }
    },
    render: {
      value: function render() {
        return React.createElement(
          "div",
          { style: this.props.style },
          React.createElement("div", { ref: "target" }),
          React.createElement(
            "div",
            null,
            this.props.children,
            this.props.view
          )
        );
      }
    },
    focus: {
      value: function focus() {
        var viewport = this.map.getViewport();
        viewport.tabIndex = 0;
        viewport.focus();
      }
    }
  });

  return Map;
})(React.Component);

module.exports = Map;

Map.propTypes = {
  loadTilesWhileAnimating: React.PropTypes.bool,
  loadTilesWhileInteracting: React.PropTypes.bool,
  onSingleClick: React.PropTypes.func,
  onChangeSize: React.PropTypes.func,
  view: React.PropTypes.element.isRequired,
  useDefaultInteractions: React.PropTypes.bool.isRequired,
  useDefaultControls: React.PropTypes.bool.isRequired,
  focusOnMount: React.PropTypes.bool.isRequired,

  children: React.PropTypes.oneOfType([React.PropTypes.arrayOf(React.PropTypes.element), React.PropTypes.element])
};

Map.defaultProps = {
  useDefaultInteractions: true,
  useDefaultControls: true,
  focusOnMount: false
};

Map.childContextTypes = {
  map: React.PropTypes.instanceOf(ol.Map)
};
