// ol-react/lib/style.js
"use strict";

var _interopRequire = function (obj) { return obj && obj.__esModule ? obj["default"] : obj; };

exports.buildStyle = buildStyle;
Object.defineProperty(exports, "__esModule", {
  value: true
});

var ol = _interopRequire(require("openlayers"));

var STYLE_KEY_FACTORIES = {
  geometry: function (value) {
    return new ol.style.Geometry(value);
  },
  fill: function (value) {
    return new ol.style.Fill(value);
  },
  image: function (value) {
    return new ol.style.Image(value);
  },
  stroke: function (value) {
    return new ol.style.Stroke(value);
  },
  text: function (value) {
    return new ol.style.Text(value);
  },
  zIndex: function (value) {
    return value;
  }
};

function buildStyle(style) {
  if (!style) {
    return null;
  }

  if (Array.isArray(style)) {
    return style.map(buildStyle);
  }

  if (typeof style === "function") {
    return style;
  }

  if (ol.style.Style.prototype.isPrototypeOf(style)) {
    return style;
  }

  var result = {};

  Object.keys(STYLE_KEY_FACTORIES).filter(function (key) {
    return !!style[key];
  }).forEach(function (key) {
    result[key] = STYLE_KEY_FACTORIES[key](style[key]);
  });

  return new ol.style.Style(result);
}
