// ol-react/lib/control/zoom.js
"use strict";

var _interopRequire = function (obj) { return obj && obj.__esModule ? obj["default"] : obj; };

var _createClass = (function () { function defineProperties(target, props) { for (var key in props) { var prop = props[key]; prop.configurable = true; if (prop.value) prop.writable = true; } Object.defineProperties(target, props); } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; })();

var _inherits = function (subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) subClass.__proto__ = superClass; };

var _classCallCheck = function (instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } };

var React = _interopRequire(require("react"));

var ol = _interopRequire(require("openlayers"));

var OLControl = _interopRequire(require("./ol-control"));

var Zoom = (function (_OLControl) {
  function Zoom() {
    _classCallCheck(this, Zoom);

    if (_OLControl != null) {
      _OLControl.apply(this, arguments);
    }
  }

  _inherits(Zoom, _OLControl);

  _createClass(Zoom, {
    createControl: {
      value: function createControl(props) {
        return new ol.control.Zoom({
          className: props.className,
          delta: props.delta,
          duration: props.duration,
          zoomInLabel: props.zoomInLabel,
          zoomInTipLabel: props.zoomInTipLabel,
          zoomOutLabel: props.zoomOutLabel,
          zoomOutTipLabel: props.zoomOutTipLabel
        });
      }
    }
  });

  return Zoom;
})(OLControl);

module.exports = Zoom;

Zoom.propTypes = Object.assign({}, OLControl.propTypes, {
  className: React.PropTypes.string,
  delta: React.PropTypes.number,
  duration: React.PropTypes.number,
  zoomInLabel: React.PropTypes.node,
  zoomInTipLabel: React.PropTypes.string,
  zoomOutLabel: React.PropTypes.node,
  zoomOutTipLabel: React.PropTypes.string
});
