// ol-react/lib/interaction/keyboard-zoom.js
"use strict";

var _interopRequire = function (obj) { return obj && obj.__esModule ? obj["default"] : obj; };

var _createClass = (function () { function defineProperties(target, props) { for (var key in props) { var prop = props[key]; prop.configurable = true; if (prop.value) prop.writable = true; } Object.defineProperties(target, props); } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; })();

var _inherits = function (subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) subClass.__proto__ = superClass; };

var _classCallCheck = function (instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } };

var React = _interopRequire(require("react"));

var ol = _interopRequire(require("openlayers"));

var OLInteraction = _interopRequire(require("./ol-interaction"));

var KeyboardZoom = (function (_OLInteraction) {
  function KeyboardZoom() {
    _classCallCheck(this, KeyboardZoom);

    if (_OLInteraction != null) {
      _OLInteraction.apply(this, arguments);
    }
  }

  _inherits(KeyboardZoom, _OLInteraction);

  _createClass(KeyboardZoom, {
    createInteraction: {
      value: function createInteraction(props) {
        return new ol.interaction.KeyboardZoom({
          condition: props.condition,
          delta: props.delta,
          duration: props.duration
        });
      }
    }
  });

  return KeyboardZoom;
})(OLInteraction);

module.exports = KeyboardZoom;

KeyboardZoom.propTypes = Object.assign({}, OLInteraction.propTypes, {
  condition: React.PropTypes.func,
  delta: React.PropTypes.number,
  duration: React.PropTypes.number
});
